SPEC/BITSWAP.md

```diff
@@ -5,60 +5,59 @@
 
 ### `bitswap.wantlist`
 
-> Returns the wantlist, optionally filtered by peer ID
+> Get the list of wanted CIDs for this peer or another peer on the network.
 
-#### `Go` **WIP**
+#### Go **WIP**
 
-#### `JavaScript` - ipfs.bitswap.wantlist([peerId], [callback])
+#### JavaScript - `ipfs.bitswap.wantlist([peerId])`
 
-`callback` must follow `function (err, list) {}` signature, where `err` is an error if the operation was not successful. `list` is an Object containing the following keys:
+##### Parameters
 
-- `Keys` An array of objects containing the following keys:
-    - `/` A string multihash
+| Name | Type | Description |
+|------|------|-------------|
+| peerId | `String` | (optional) Base 58 encoded Peer ID to get the wantlist for. Default: this node's peer ID |
 
-If no `callback` is passed, a promise is returned.
+##### Returns
 
-**Example:**
+| Type | Description |
+|------|-------------|
+| `Promise<{`<br/>&nbsp;&nbsp;`Keys<Array<Object>>`<br/>`}>` | The list of CIDs wanted by the peer. Each object in the array has a single property "/" a string CID. |
 
-```JavaScript
-ipfs.bitswap.wantlist((err, list) => console.log(list))
+##### Example
 
-// { Keys: [{ '/': 'QmHash' }] }
+```js
+const list = await ipfs.bitswap.wantlist()
```

Comment:
Should this return an async iterable instead of a list?

Reply:
I think in general we should either return single values or iterators of single values, never lists of single values.

Reply:
I agree, the wantlist could get really big so an iterator makes sense.

> never lists of single values

Do you think this should always be true? I think we should return async iterators where it makes sense to "stream" the output but I wouldn't have thought it would be necessary for something like `ipfs.bootstrap.list`, for example...but could be persuaded for the sake of consistency.

Reply:
we talking iterable ? or really async iterator ?